include/secp256k1_silentpayments.h

```diff
@@ -25,6 +26,99 @@ extern "C" {
  *  any further elliptic-curve operations from the wallet.
  */
 
+
+/** The data from a single recipient address
+ *
+ *  This struct serves as an input argument to `silentpayments_sender_create_outputs`.
+ *
+ *  `index` must be set to the position (starting with 0) of this recipient in the
+ *  `recipients` array passed to `silentpayments_sender_create_outputs`. It is
+ *  used to map the returned generated outputs back to the original recipient.
+ *
+ *  Note:
+ *  The spend public key named `spend_pubkey` may have been optionally tweaked with
+ *  a label by the recipient. Whether `spend_pubkey` has actually been tagged with
+ *  a label is irrelevant for the sender. As a documentation convention in this API,
+ *  `unlabeled_spend_pubkey` is used to indicate when the unlabeled spend public must
+ *  be used.
+ */
+typedef struct secp256k1_silentpayments_recipient {
+    secp256k1_pubkey scan_pubkey;
+    secp256k1_pubkey spend_pubkey;
+    size_t index;
+} secp256k1_silentpayments_recipient;
+
+/** Create Silent Payment outputs for recipient(s).
+ *
+ *  Given a list of n secret keys a_1...a_n (one for each silent payment
+ *  eligible input to spend), a serialized outpoint, and a list of recipients,
+ *  create the taproot outputs. Inputs with conditional branches or multiple
+ *  public keys are excluded from silent payments eligible inputs; see BIP352
+ *  for more information.
+ *
+ *  `outpoint_smallest36` refers to the smallest outpoint lexicographically
+ *  from the transaction inputs (both silent payments eligible and non-eligible
+ *  inputs). This value MUST be the smallest outpoint out of all of the
+ *  transaction inputs, otherwise the recipient will be unable to find the
+ *  payment. Determining the smallest outpoint from the list of transaction
+ *  inputs is the responsibility of the caller. It is strongly recommended
+ *  that implementations ensure they are doing this correctly by using the
+ *  test vectors from BIP352.
+ *
+ *  Returns: 1 if creation of outputs was successful.
+ *           0 if the input private keys sum to zero,
+ *             or input_hash or hash(shared_secret || k) are invalid scalars.
+ *  Args:                ctx: pointer to a context object
+ *                            (not secp256k1_context_static).
+ *  Out:   generated_outputs: pointer to an array of pointers to xonly public keys,
+ *                            one per recipient.
+ *                            The outputs are ordered to match the original
+ *                            ordering of the recipient objects, i.e.,
+ *                            `generated_outputs[0]` is the generated output
+ *                            for the `_silentpayments_recipient` object with
+ *                            index = 0.
```

Comment:
In 8044d3770cb64392ad9b93ad7f44da66e92cf159 _silentpayments: sending_: in line with our earlier discussion, it might be worth warning the user to not drop outputs (at least not for transactions with more than one recipient).

https://github.com/bitcoin-core/secp256k1/pull/1698/commits/8044d3770cb64392ad9b93ad7f44da66e92cf159#r2245582298

Reply:
Added!